docs/adr/adr-0026-introducing-prettier.md

```diff
@@ -0,0 +1,176 @@
+
+### dprint
+
+From [dprint's website](https://dprint.dev/):
+
+> A pluggable and configurable code formatting platform written in Rust.
+
+#### Pros
+
+- Lot of options.
+- You can develop plugins.
+- If you want to develop plugins, there's a rust SDK.
+
+#### Cons
+
+- Even though it's possible to write plugins in another language, resources about this are sparse. It means you better be proficient with Rust.
+
+### Prettier
+
+[Prettier](https://prettier.io) is an opiniated code formatter.
+
+#### Pros
+
+- Can format both JS, CSS, template literals in one take.
+
+#### Cons
+
+- There's not a lot of options.
+  - It is justified by the `opiniated formatter` logic.
+
+Prettier is the option we chose to use, you can find more information about the why on the dedicated section below.
+
+### js-beautify
+
+#### Pros
+
+- Has integrations with IDEs
+
+#### Cons
+
+- Unsure about the future of the maintenance ([See github README](https://github.com/beautifier/js-beautify?tab=readme-ov-file#contributors-needed))
+- Can over-format
+
+### Others
+
+- While we knew all these tools existed we wanted to know what Lit users were using
+  - Knowing that we asked the Lit discord's what they were using
+    - Someone told us that they were using the new `Stylistic` package for ESLint that keeps formatting rules
+    - They were also using a Prettier plugin for CSS
+    - While it seemed a good idea at first, you'll find why it wasn't adapted to our case below.
+
+## Investigation after Lit Discord's question (TODO: remove this)
+
+- I have asked the Lit Discord's to know if some users used specific tools to format their Lit components
+  - It seems like they're using a `Stylistic` package that should maintains the deprecated rules
+  - However, as the new ESLint version is quite new (04-05-2024) there are a lot of packages which are not working on version 9
+    - I suggest that we don't migrate to version 9 until some of the plugins are updated or that we find another formatter
+  - About Stylelint:
+    - Version 15 deprecated its formatting rules, while removing them from version 16
+    - Processors have been removed and changed to `customSyntax`
+      - You can have multiple of them for different purposes
+        - e.g: `postcss-lit` for `.js` component files
+        - e.g: `postcss` for `.css` files
+      - This introduction made the Stylelint `postcss-css-in-js` processor/plugin obsolete.
+        - Luckily there's now a `customSyntax` for
+
+## Solution
+
+### Why did we choose Prettier?
+
+We decided to go for Prettier for various reasons:
+
+- It is widely used and maintained.
+- Has integration with IDEs.
+- It enforces code style. Beside our internal conventions we didn't have a proper formatter enforcing.
+
+Note: As we didn't update ESLint yet, see details on the dedicated section below, we added the Prettier ESLint plugin so that ESLint formatting rules doesn't conflict with Prettier.
```

Comment:
**nitpick:**
```suggestion
Note: As we haven't updated ESLint yet, see details on the dedicated section below, we've added the Prettier ESLint plugin so that ESLint formatting rules doesn't conflict with Prettier.
```